testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -52,70 +72,149 @@ impl NetworkTest for StateSyncPerformance {
-                "State sync TPS requirement failed. Average TPS: {}, minimum required TPS: {}",
-                state_sync_throughput, min_expected_tps
-            );
-            bail!(error_message)
+        // Generate some traffic through the validators.
+        let emit_txn_duration = ctx.global_duration.checked_div(2).unwrap();
+        info!(
+            "Generating the first round of traffic for {:?} seconds.",
+            emit_txn_duration.as_secs()
+        );
+        let _txn_stat = generate_traffic(ctx, &all_validators, emit_txn_duration, 1)?;
+
+        // Wait for all nodes to synchronize and stabilize.
+        info!("Waiting for the validators to be synchronized.");
+        let runtime = Runtime::new().unwrap();
+        runtime.block_on(async {
+            ctx.swarm()
+                .wait_for_all_nodes_to_catchup(Duration::from_secs(MAX_NODE_LAG_SECS))
+                .await
+        })?;
+
+        // Stop and reset two validators
+        info!("Deleting data for two validators!");
+        let validators_to_restart = &all_validators[0..2];
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.clear_storage().await })?;
+        }
+
+        // Restart the validators so they start syncing from a fresh state
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.start().await })?;
         }
 
-        Ok(())
+        // Generate some additional traffic through the validators so the fresh
+        // validators are forced to try and catch up to a moving target.
+        info!(
+            "Generating the second round of traffic for {:?} seconds.",
+            emit_txn_duration.as_secs()
+        );
+        let _txn_stat = generate_traffic(ctx, &all_validators, emit_txn_duration, 1)?;
+
+        // Wait for all nodes to catch up to the highest synced version
+        // then calculate and display the throughput results.
+        calculate_and_display_throughput(ctx, self.name())
     }
 }
+
+/// Calculates and displays the state sync throughput using
+/// the synced version and sync duration.
+fn calculate_and_display_throughput(ctx: &mut NetworkContext<'_>, test_name: &str) -> Result<()> {
+    // Wait for all nodes to catch up to the same synced version
+    let runtime = Runtime::new().unwrap();
+    let timer = Instant::now();
+    let node_sync_duration = ctx.global_duration.saturating_mul(2); // Limits state sync to half consensus throughput
+    runtime.block_on(async {
+        ctx.swarm()
+            .wait_for_all_nodes_to_catchup(node_sync_duration)
+            .await
+    })?;
+    let seconds_to_sync = timer.elapsed().as_secs();
+
+    // Calculate the state sync throughput
+    let highest_synced_version = runtime.block_on(async {
```

Comment:
while you are waiting - chain is progressing?
do you want to take this value before wait_for_all_nodes_to_catchup?

also why highset here, not lowest (as the worst state sync throughput)? 

(get_clients_with_names returns both validators and full nodes)

Reply:
1. Indeed, this is not super ideal, but I've made it so that `highest_synced_version` is taken first. It does mean that we'll end up under-reporting state sync throughput because it's possible that `wait_for_all_nodes_to_catchup` gets all nodes to sync to a slightly higher target, but that's fine.
2. Can you clarify what you mean by `also why highset here, not lowest (as the worst state sync throughput)?`. I'm not following 🤔